src/test/java/nextstep/subway/LineAcceptanceTest.java

```diff
@@ -29,90 +30,175 @@
-        Map<String, Object> params = new HashMap<>();
-        params.put("name", 수정될_이름);
-        params.put("color", 수정될_컬러);
+    /**
+     * Given 지하철 노선을 생성하고
+     * When 하행역이 노선 상행 종점인 구간을 등록하면
+     * Then 구간의 상행역이 노선 상행 종점역으로 등록된다
+     */
+    @DisplayName("새로운 역을 상행 종점에 등록한다")
+    @Test
+    void insertSectionToUpTerminalStationSuccess() {
+        // given
+        int 강남역_아이디 = 아이디(지하철역을_생성한다("강남역"));
+        int 판교역_아이디 = 아이디(지하철역을_생성한다("판교역"));
+        int 교대역_아이디 = 아이디(지하철역을_생성한다("교대역"));
+        var 신분당선 = 지하철_노선_등록한다(
+                "신분당선",
+                "bg-red-600",
+                강남역_아이디,
+                판교역_아이디,
+                10);
 
-        return RestAssured
-                .given().accept(ContentType.JSON).contentType(ContentType.JSON).body(params)
-                .when().put("/lines/" + 수정될_지하철_노선_아이디)
-                .then().log().all()
-                .extract();
-    }
+        // when
+        노선_구간을_등록한다(아이디(신분당선), 교대역_아이디, 강남역_아이디, 1);
 
-    private static ExtractableResponse<Response> 지하철_노선_삭제한다(int 신분당선_아이디) {
-        return RestAssured
-                .given().log().all()
-                .when().delete("/lines/" + 신분당선_아이디)
-                .then().log().all()
-                .extract();
+        // then
+        신분당선 = 지하철_노선_조회한다(아이디(신분당선));
+        List<String> 스테이션_이름_리스트 = 스테이션_이름_리스트(신분당선);
+        assertThat(스테이션_이름_리스트).containsExactly("교대역", "강남역", "판교역");
     }
 
-    private static ExtractableResponse<Response> 노선_구간을_등록한다(int 노선_아이디, int 상행역, int 하행역,
-            int 거리) {
-        Map<String, Object> params = new HashMap<>();
-        params.put("downStationId", 하행역);
-        params.put("upStationId", 상행역);
-        params.put("distance", 거리);
-        return RestAssured
-                .given().body(params).log().all()
-                .contentType(ContentType.JSON).accept(ContentType.JSON)
-                .when().post("/lines/" + 노선_아이디 + "/sections")
-                .then().log().all()
-                .extract();
-    }
+    /**
+     * Given 지하철 노선을 생성하고
+     * When 상행역이 기존 구간의 상행역과 일치하고 똑같이 않은 구간을 등록하면
+     * Then 기존 구간의 역을 기준으로 새로운 구간을 추가된다
+     */
+    @DisplayName("기존 구간의 역을 기준으로 새로운 구간을 추가")
+    @Test
+    void insertSectionToCenterSuccess() {
+        // given
+        int 강남역_아이디 = 아이디(지하철역을_생성한다("강남역"));
+        int 판교역_아이디 = 아이디(지하철역을_생성한다("판교역"));
+        int 양재역_아이디 = 아이디(지하철역을_생성한다("양재역"));
+        var 신분당선 = 지하철_노선_등록한다(
+                "신분당선",
+                "bg-red-600",
+                강남역_아이디,
+                판교역_아이디,
+                10);
 
-    private static ExtractableResponse<Response> 노선에서_구간_제거한다(int 종점역_아이디, int 노선_아이디) {
-        return RestAssured
-                .given().log().all().param("stationId", 종점역_아이디)
-                .when().delete("/lines/" + 노선_아이디 + "/sections")
-                .then().log().all()
-                .extract();
-    }
+        // when
+        노선_구간을_등록한다(아이디(신분당선), 강남역_아이디, 양재역_아이디, 1);
```

Comment:
```suggestion
        노선_구간을_등록한다(아이디(신분당선), 강남역_아이디, 양재역_아이디, 1);
       
        int 선릉역_아이디 = 아이디(지하철역을_생성한다("선릉역"));
        int 역삼역_아이디 = 아이디(지하철역을_생성한다("역삼역"));
        노선_구간을_등록한다(아이디(신분당선), 양재역_아이디, 역삼역_아이디, 3);
        노선_구간을_등록한다(아이디(신분당선), 양재역_아이디, 선릉역_아이디, 1);
```

위와 같이 추가하면 `강남역-양재역-선릉역-역삼역-판교역` 으로 정렬되어야 합니다 😅 
확인 후 수정해주세요 🔥 

<img width="534" alt="스크린샷 2023-07-19 오전 12 49 55" src="https://github.com/next-step/atdd-subway-path/assets/17218212/0aee435a-3606-41e6-8c7d-92661012455c">